test/byContract/IexecPoco/IexecPoco1.test.ts

```diff
@@ -118,14 +124,59 @@ describe('IexecPoco1', () => {
             .connect(anyone)
             .deploy()
             .then((contract) => contract.deployed());
+        erc1271Contract = await new ERC1271Mock__factory()
+            .connect(anyone)
+            .deploy()
+            .then((contract) => contract.deployed());
         // TODO check why this is done in 00_matchorders.js
         // await Workerpool__factory.connect(workerpoolAddress, scheduler)
         //     .changePolicy(35, 5)
         //     .then((tx) => tx.wait());
     }
 
+    describe('Verify signature', () => {
+        ['verifySignature', 'verifyPresignatureOrSignature'].forEach((verifySignatureFunction) => {
+            it(`Should ${verifySignatureFunction} of smart contract`, async () => {
+                expect(
+                    await iexecPocoContract[verifySignatureFunction](
+                        erc1271Contract.address,
+                        HashZero, // any is fine here
+                        ethers.utils.id('valid-signature'),
+                    ),
+                ).to.be.true;
+            });
+            it(`Should fail to ${verifySignatureFunction} of smart contract when validation returns false`, async () => {
+                expect(
+                    await iexecPocoContract[verifySignatureFunction](
+                        erc1271Contract.address,
+                        HashZero, // any is fine here
+                        ethers.utils.id('invalid-signature'),
+                    ),
+                ).to.be.false;
+            });
+            it(`Should fail to ${verifySignatureFunction} of smart contract when validation reverts`, async () => {
+                expect(
+                    await iexecPocoContract[verifySignatureFunction](
+                        erc1271Contract.address,
+                        HashZero, // any is fine here
+                        ethers.utils.id('reverting-signature'),
+                    ),
+                ).to.be.false;
+            });
+            it(`Should ${verifySignatureFunction} of EoA`, async () => {
+                const wallet = ethers.Wallet.createRandom();
+                const someMessage = 'some-message';
+                expect(
+                    await iexecPocoContract[verifySignatureFunction](
+                        wallet.address,
+                        ethers.utils.hashMessage(someMessage),
+                        wallet.signMessage(someMessage),
+                    ),
+                ).to.be.true;
+            });
+        });
+    });
     // TODO
-    describe('Verify signature', () => {});
```

Comment:
Let's add test for EOA signature with 64 bytes and for invalid signature format or keep the TODO because we don't cover all cases right now.

Reply:
Thank you, applied suggestions here: [Add EOA signature tests](https://github.com/iExecBlockchainComputing/PoCo/pull/120/commits/4d372029c18b41bcc0ad616e19ddbcfeb16df871) :+1: